linter.py

```diff
@@ -51,16 +59,48 @@ class Flow(Linter):
 
     def cmd(self):
         """Return the command line to execute."""
-        command = [self.executable_path, 'check']
 
-        if self.get_merged_settings()['show-all-errors']:
-            command.append('--show-all-errors')
+        command = [self.executable_path]
+
+        # Overrides 'use-server' directive if pressent
+        if not self.get_merged_settings()['use-server']:
+            command.append('check')
+        else:
+            if self.get_merged_settings()['lint-view']:
+                command.append('check-contents')
+            else:
+                command.append('status')
 
         if self.get_merged_settings()['all']:
-            command.append('--all')
+            if 'check' in command or 'status' in command:
+                command.append('--all')
+
+        # TODO: Remove the parameter if deemed unececerry.
+        # if self.get_merged_settings()['lib'] and len(self.get_merged_settings()['lib']) > 0:
+        #    command.append('--lib')
+        #    command.append(self.get_merged_settings()['lib'])
+
+        if self.get_merged_settings()['show-all-errors']:
+            command.append('--show-all-errors')
 
         return command
 
+    def run(self, cmd, code):
```

Comment:
The linter shouldn't check for the `@flow`, that's flow's responsibility.

Your correct that it lints the file. Until the best flow argument gets shaken out the proper solution would probably be to make it a [file only linter](http://www.sublimelinter.com/en/latest/linter_attributes.html#file-only-linters) using `tempfile_suffix = '-'`